content/documentation/runtimes-monitoring/index.adoc

```diff
@@ -0,0 +1,228 @@
+
+=== Node.js
+
+Contains some metrics such as active handles, event loop lag and heap usage. The expected metrics are provided by link:https://www.npmjs.com/package/prom-client[prom-client].
+
+Example of Node.js metrics for Prometheus:
+
+```javascript
+const client = require('prom-client');
+client.collectDefaultMetrics();
+// ...
+app.get('/metrics', (request, response) => {
+  response.set('Content-Type', client.register.contentType);
+  response.send(client.register.metrics());
+});
+```
+
+Full working example: https://github.com/jotak/bookinfo-runtimes/tree/master/ratings
+
+The pod annotation for Kiali is: `kiali.io/runtimes: nodejs`
+
+=== Spring Boot
+
+Two dashboards are provided: one for JVM metrics and another for Tomcat metrics. The expected metrics come from link:https://docs.spring.io/spring-boot/docs/current/reference/html/production-ready-metrics.html#production-ready-metrics-export-prometheus[Spring Boot Actuator for Prometheus]. Example with maven:
+
+```xml
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-actuator</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>io.micrometer</groupId>
+      <artifactId>micrometer-core</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>io.micrometer</groupId>
+      <artifactId>micrometer-registry-prometheus</artifactId>
+    </dependency>
+```
+
+Full working example: https://github.com/jotak/bookinfo-runtimes/tree/master/details
+
+The pod annotation for Kiali with the full list of dashboards is: `kiali.io/runtimes: springboot-jvm,springboot-tomcat`
+
+By default, the metrics are exposed on path _/actuator/prometheus_, so it must be specified in the corresponding annotation: `prometheus.io/path: "/actuator/prometheus"`
+
+== Create new dashboards
+
+The default dashboards described above are just examples of what we can have. It's pretty easy to create new dashboards.
+
+When installing Kiali, a new CRD is installed in the system: _monitoringdashboard.monitoring.kiali.io_. It declares the resource kind _MonitoringDashboard_. Here's how this resource looks like:
+
+```yaml
+apiVersion: "monitoring.kiali.io/v1alpha1"
+kind: MonitoringDashboard
+metadata:
+  name: vertx-server
+spec:
+  runtime: Vert.x
+  title: Vert.x Server Metrics
+  charts:
+    - name: "Server response time"
+      unit: "s"
+      spans: 6
+      metricName: "vertx_http_server_responseTime_seconds"
+      dataType: "histogram"
+      aggregations:
+        - label: "path"
+          displayName: "Path"
+        - label: "method"
+          displayName: "Method"
+    - name: "Server active connections"
+      unit: ""
+      spans: 6
+      metricName: "vertx_http_server_connections"
+      dataType: "raw"
+```
+
+The *name* field (from metadata) corresponds to what you will have to set in pods annotation `kiali.io/runtimes`.
```

Comment:
Maybe add a link on the word "kiali.io/runtimes" back up to where you described this in 2. Pods Annotations?   Also it might be helpful to have this example match the example in the Pods Annotations (right now one is Thorntail and the other is vertx).

Reply:
Ok, I keep the vertx example as it is more complete (includes more different types of data)